test/Common.FunctionalTests/Utilities/Helpers.cs

```diff
@@ -64,12 +65,15 @@ public static void CopyFiles(DirectoryInfo source, DirectoryInfo target, ILogger
         {
             foreach (DirectoryInfo directoryInfo in source.GetDirectories())
             {
-                CopyFiles(directoryInfo, target.CreateSubdirectory(directoryInfo.Name), logger);
+                if (directoryInfo.FullName != target.FullName)
```

Comment:
Why was this check needed?

Reply:
To avoid infinite recursion when directory it's copied into it's subdirectory.